provider/pkg/provider/provider.go

```diff
@@ -121,17 +121,33 @@ func (p *googleCloudProvider) Configure(ctx context.Context,
 	// * GCLOUD_PROJECT Env Var
 	// * CLOUDSDK_CORE_PROJECT Env Var
 	// This aligns with our provider documentation
-	project := p.getConfig("project", []string{
+	projectValue := p.getConfig("project", []string{
 		"GOOGLE_PROJECT",
 		"GOOGLE_CLOUD_PROJECT",
 		"GCLOUD_PROJECT",
 		"CLOUDSDK_CORE_PROJECT",
 	})
-	if project == "" {
-		return nil, fmt.Errorf("unable to find a valid Project. Set the Project by using: \n\n" +
-			" \t • `pulumi config set google-native:project my-project` or \n" +
-			" \t • Environment variables: `GOOGLE_PROJECT`, `GOOGLE_CLOUD_PROJECT`, `GCLOUD_PROJECT` or `CLOUDSDK_CORE_PROJECT` \n\n")
+
+	if projectValue != "" {
+		p.config["project"] = projectValue
```

Comment:
What happens if `p.config["project"]` isn't set? Will the provider to continue looking for the Project on a per-resource basis?

Reply:
Yes, and they will receive a helpful error during the preview on the first resource that is missing the information.

Reply:
I believe it should - see https://github.com/pulumi/pulumi-google-native/blob/430dbca00c6c20ce59670ddafc0bafa720cc5127/provider/pkg/provider/provider.go#L323

However https://github.com/pulumi/pulumi-google-native/issues/565 seems to suggest it doesn't. Can we validate against the resource in that report or other resources?

Reply:
This error is correct:
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/41454626/187798399-abb34890-e542-4a42-a488-770a6d1680cd.png">
but could perhaps be handled better in the api docs, indicating it would be required if not set via other means:
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/41454626/187798479-d9d8aede-780a-44ea-b364-74a0998e787c.png">

Reply:
Yeah can you open a separate issue? We need to update the docs anyway since the explicit requirement on project is no longer accurate: https://www.pulumi.com/registry/packages/google-native/installation-configuration/#configuration-options

Reply:
I will update the guide as I have: https://github.com/pulumi/registry/issues/1418 in progress and I will make a separate issue about the api docs if it doesn't already exist, thank you!